### DISCUSSION TITLE
Data manipulation on migration

### DISCUSSION THREAD

**Discussion by seyedmmousavi** [ent/ent#3117] [Q&A]
Hi dear,  
Thank 's for great production the Ent!
Is it possible to migrate data itself between schema versions in Ent?  
It is useful to keep data compatibility in some scenarios.

**Comment by a8m** [comment]
Hey @seyedmmousavi, and thanks for raising this point. I opened a PR with docs+examples for this question. See: https://github.com/ent/ent/pull/3119

You can also read the complete doc on our website, see: https://entgo.io/docs/data-migrations

**Reply by seyedmmousavi** [reply]
OK,  
I read it and it solves several problems very good. But, what if I need to process data (some or all records) with some application logic? It might be because the migration motivation is changes in logic. In such scenarios the data must be loaded, processed and saved again. This is what I faced in agile, fast growing business with uncertain logics like startup.  
  
The mentioned PR is not present in my current clone. 
```
gh pr checkout 3119
fatal: couldn't find remote ref refs/heads/datamigrate
failed to run git: exit status 128
```

**Reply by a8m** [reply]
> The mentioned PR is not present in my current clone.

Use Go for this: `go get entgo.io/ent@master`.

>  In such scenarios the data must be loaded, processed and saved again. This is what I faced in agile, fast growing business with uncertain logics like startup.

What type of support do you expect from Ent/Atlas in this scenario? I also have such tasks that I do with Ent, but we manage this differently because we don't treat it as migration. It's a data processing, that may be done in several chunks and need to be checkpointed or marked as completed in an external service. Also, in order to prevent incorrect mutation of data, adding threshold/sampling and verification mechanisms is also a good practice.
